runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -38,6 +38,7 @@ public class BoundedQueueExecutor {
   private int elementsOutstanding = 0;
   private long bytesOutstanding = 0;
   private final AtomicInteger activeCount = new AtomicInteger();
+  private final AtomicInteger maximumThreadCount = new AtomicInteger();
```

Comment:
would be good to add a unit test covering the new methods to this class directly

Reply:
Sure.